### PR TITLE
Connect login and register to work with backend and save token to localstorage

### PR DESCRIPTION
**Pull request by DaniilPoiarkov**
Added ability to login and register via frontend pages

### CODE REVIEW

File: backend/.gitignore [hunk 1]
```diff
@@ -9,6 +9,7 @@
 *.user
 *.userosscache
 *.sln.docstates
+*.appsettings.Development.json
```
Comment (ihor-il) [inline backend/.gitignore:12]: you can use user secrets so you don't have to gitignore this file

**Review by ihor-il — COMMENTED** [review]
Is this supposed to have service so it can actually send requests and receive tokens?

File: frontend/src/app/components/register-page/register-page.component.ts [hunk 2]
```diff
@@ -10,33 +14,36 @@ import { ValidationConstants } from 'src/entity-models/const-resources/validatio
...
   public hidePass = true;
   public hidePassRepeat = true;
+
+  public userRegister: UserRegisterModel = {};
   public registerForm: FormGroup =  new FormGroup({});
   public nameControl: FormControl;
   public emailControl: FormControl;
   public passwordControl: FormControl;
   public passwordRepeatControl: FormControl;
+  public unsubscribe$ = new Subject<void>();
```
Comment (Anatolii-Ivanov) [inline frontend/src/app/components/register-page/register-page.component.ts]: Why do you need this?

**Review by ismv — CHANGES_REQUESTED** [review]
Conflicts Resolving

File: frontend/src/app/components/register-page/register-page.component.ts (hunk 2 above)
Comment (DaniilPoiarkov) [reply frontend/src/app/components/register-page/register-page.component.ts]: In this file user sending register form to backend=>
this.service.registerUser().pipe(takeUntil(this.unsub..$)).subscribe(...)

**Comment by DaniilPoiarkov** [comment]
> Conflicts Resolving

Done

File: frontend/src/app/components/register-page/register-page.component.ts (hunk 2 above)
Comment (Anatolii-Ivanov) [reply frontend/src/app/components/register-page/register-page.component.ts]: Yes, but cannot you just subscribe?

Comment (DaniilPoiarkov) [reply frontend/src/app/components/register-page/register-page.component.ts]: > Yes, but cannot you just subscribe?
That is a question.
Done

Comment (DaniilPoiarkov) [reply frontend/src/app/components/register-page/register-page.component.ts]: > Yes, but cannot you just subscribe?

That is a question.
Done

File: frontend/src/entity-models/const-resources/validation-constraints.ts [hunk 3]
```diff
@@ -2,5 +2,5 @@ export const ValidationConstants = {
     minLengthName: 4,
     minLengthPassword: 8,
     minLengthEmail: 8,
-    emailRegex: '^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$'
+    emailRegex: '^[a-zA-Z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$'
```
Comment (ihor-il) [inline frontend/src/entity-models/const-resources/validation-constraints.ts:5]: This regex is invalid:
- `\\` after domain?
- i suggest using `^[\w-\.]+@([\w-]+\.)+[\w-]{2,3}$` instead

**Review by ihor-il — CHANGES_REQUESTED** [review]

**Review by oleksandr-vovk — APPROVED** [review]

File: frontend/src/entity-models/const-resources/validation-constraints.ts (hunk 3 above)
Comment (DaniilPoiarkov) [reply frontend/src/entity-models/const-resources/validation-constraints.ts:5]: Why this regex is invalid? You can't use "\\" symbols in domain. Just try it

**Review by ismv — APPROVED** [review]

File: frontend/src/entity-models/const-resources/validation-constraints.ts (hunk 3 above)
Comment (ihor-il) [reply frontend/src/entity-models/const-resources/validation-constraints.ts:5]: nvm, i'm stupid

**Review by ihor-il — APPROVED** [review]

**Pull request merged**